HopsanGUI/LogVariable.cpp

```diff
@@ -805,6 +805,130 @@ SharedVectorVariableT VectorVariable::toFrequencySpectrum(const SharedVectorVari
     }
 }
 
+SharedVectorVariableT VectorVariable::identifySteadyState(const SteadyStateIdentificationMethodEnumT method, double tol, double win, double stdev, double l1, double l2, double l3)
+{
+    SharedVectorVariableT pRetVar = mpParentLogDataHandler->createOrphanVariable(this->getFullVariableName()+"_ss", TimeDomainType);
+
+    SharedVectorVariableT pTime = this->getSharedTimeOrFrequencyVector();
+    QVector<double> time = pTime->getDataVectorCopy();
+    QVector<double> data = this->getDataVectorCopy();
+
+    if(method == RectangularWindowTest) {
+        int nw = 0;
+        for(int i=0; i<time.size(); ++i) {
+            if(time[i] < win) {
+                pRetVar->append(time[i], 0);
+                nw = i+1;  //Update number of samples for window
+                continue;
+            }
+
+            //Compute first variance
+            double xmax = data[i-nw];
+            double xmin = data[i-nw];
+            for(int j=i-nw+1; j<i; ++j) {
+                if(data[j] > xmax) {
+                    xmax = data[j];
+                }
+                else if(data[j] < xmin) {
+                    xmin = data[j];
+                }
+            }
+
+            if(xmax-xmin < tol) {
+                pRetVar->append(time[i], 1);
+            }
+            else {
+                pRetVar->append(time[i], 0);
+            }
+        }
+    }
+    else if(method == VarianceRatioTest) {
+
+        //Create randomized data vector
+        QVector<double> x;
+        x.resize(data.size());
+        for(int i=0; i<data.size(); ++i) {
+            x[i] = normalDistribution(data[i], stdev);
+        }
+
+        int nw = 0;
+        for(int i=0; i<time.size(); ++i) {
+            if(time[i] < win) {
+                pRetVar->append(time[i], 0);
+                nw = i+1;  //Update number of samples for window
+                continue;
+            }
+
+
+            //Compute average of window
+            double mean = 0;
+            for(int j=i-nw; j<i; ++j) {
```

Comment:
Same here for this and the following for loop. They will become slower and slower as j grows.
Maybe it has to be this way?